### PR TITLE
test(select): refine testing suite

### PR DESCRIPTION
**Pull request by paulrobinson88**
### Proposed behaviour
- Removes unnecessary or duplicated tests from the Playwright file for `simple-select`, `filterable-select` and `multi-select`.
- Disables Chromatic snapshots which aren't required.
- Renames some of the test stories to remove the select variant name.

### Current behaviour
- There are a number of tests in the Playwright file which are already covered elsewhere in the testing suite.
- Some Chromatic snapshots are being taken which aren't required
- Test stories for the different versions of `Select` have the type as part of the name such as `Filterable Select With Lazy Loading Component`  

### Checklist

- [ ] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions
- Ensure Playwright tests pass as expected
- Chromatic snapshots are correct
- Stories in the `Test` folder have the correct naming

### CODE REVIEW

File: src/components/select/filterable-select/filterable-select.pw.tsx [hunk 1]
```diff
@@ -34,22 +34,14 @@ import {
...
   selectOptionByText,
 } from "../../../../playwright/components/select";
-import {
-  checkAccessibility,
-  positionOfElement,
-} from "../../../../playwright/support/helper";
-import { CHARACTERS } from "../../../../playwright/support/constants";
+import { checkAccessibility } from "../../../../playwright/support/helper";
 
-const testData = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
```
Comment (sianford) [inline src/components/select/filterable-select/filterable-select.pw.tsx:46]: Why are we removing these tests? These are not covered in unit test and only alphanumeric values are captured by Chromatic in the stories.

File: src/components/select/filterable-select/filterable-select.pw.tsx [hunk 2]
```diff
@@ -85,247 +77,6 @@ test.describe("FilterableSelect component", () => {
     await expect(getDataElementByValue(page, "input")).toHaveValue("f");
   });
 
-  testData.forEach((labelValue) => {
```
Comment (sianford) [inline src/components/select/filterable-select/filterable-select.pw.tsx:88]: As above, RE special character and diacritics tests.

File: src/components/select/filterable-select/filterable-select.pw.tsx [hunk 3]
```diff
@@ -85,247 +77,6 @@ test.describe("FilterableSelect component", () => {
...
-    test(`should render label using ${labelValue} special characters`, async ({
```
Comment (sianford) [inline src/components/select/filterable-select/filterable-select.pw.tsx:89]: As above, RE special character and diacritics tests.

File: src/components/select/filterable-select/filterable-select.pw.tsx [hunk 4]
```diff
@@ -85,247 +77,6 @@ test.describe("FilterableSelect component", () => {
...
-      mount,
-      page,
-    }) => {
-      await mount(<FilterableSelectComponent label={labelValue} />);
-
-      await expect(getDataElementByValue(page, "label")).toHaveText(labelValue);
-    });
-  });
-
-  testData.forEach((placeholderValue) => {
```
Comment (sianford) [inline src/components/select/filterable-select/filterable-select.pw.tsx:99]: As above, RE special character and diacritics tests.

File: src/components/select/filterable-select/filterable-select.pw.tsx [hunk 5]
```diff
@@ -85,247 +77,6 @@ test.describe("FilterableSelect component", () => {
...
-    test(`should render placeholder using ${placeholderValue} special characters`, async ({
```
Comment (sianford) [inline src/components/select/filterable-select/filterable-select.pw.tsx:100]: As above, RE special character and diacritics tests.

File: src/components/select/filterable-select/filterable-select.pw.tsx [hunk 6]
```diff
@@ -85,247 +77,6 @@ test.describe("FilterableSelect component", () => {
...
-    );
-  });
-
-  test("should check autofocus prop", async ({ mount, page }) => {
-    await mount(<FilterableSelectComponent autoFocus />);
-
-    await expect(commonDataElementInputPreview(page)).toBeFocused();
-  });
-
-  test("when maxWidth has no value it should render as 100%", async ({
```
Comment (sianford) [inline src/components/select/filterable-select/filterable-select.pw.tsx:177]: I think this should be added as a unit test. There's a test for when `maxWidth` has a value, when it's not passed at all, but not if it's passed with an empty value.

File: src/components/select/filterable-select/filterable-select.pw.tsx [hunk 7]
```diff
@@ -802,111 +484,6 @@ test.describe("FilterableSelect component", () => {
...
-  }) => {
-    await mount(<FilterableSelectMultiColumnsComponent />);
-
-    await dropdownButton(page).click();
-    await expect(
-      multiColumnsSelectListBody(page).locator(".."),
-    ).toHaveAttribute("data-role", "option-row");
-  });
-
-  test("should render option row data-element prop set to option-row", async ({
```
Comment (sianford) [inline src/components/select/filterable-select/filterable-select.pw.tsx:853]: This is not tested in unit testing. It should be added as an additional unit test.

File: src/components/select/filterable-select/filterable-select.stories.tsx [hunk 8]
```diff
@@ -27,6 +27,12 @@ const meta: Meta<typeof FilterableSelect> = {
   argTypes: {
     ...styledSystemProps,
   },
+  parameters: {
```
Comment (sianford) [inline src/components/select/filterable-select/filterable-select.stories.tsx:30]: This will mean that the only examples captured by Chromatic will be those in Interactions tests, is that intended? I mean, I think it covers what's needed. If that was intended, based on the last snapshots taken from master, the 'No results' snapshot looks wrong. Could you have a check of that please?

File: src/components/select/multi-select/multi-select.pw.tsx [hunk 9]
```diff
@@ -1,399 +1,29 @@
...
   selectOptionByText,
 } from "../../../../playwright/components/select";
-import {
-  checkAccessibility,
-  positionOfElement,
-  verifyRequiredAsteriskForLabel,
-} from "../../../../playwright/support/helper";
-import { CHARACTERS } from "../../../../playwright/support/constants";
-
-const testData = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
```
Comment (sianford) [inline src/components/select/multi-select/multi-select.pw.tsx:52]: These should be left in as they're not test in unit testing.

File: src/components/select/multi-select/multi-select.pw.tsx [hunk 10]
```diff
@@ -1,399 +1,29 @@
...
-  "ArrowRight",
-  "Home",
-  "End",
-  "Enter",
-] as const;
+import { checkAccessibility } from "../../../../playwright/support/helper";
 
 test.describe("MultiSelect component", () => {
-  testData.forEach((labelValue) => {
-    test(`should render label using ${labelValue} special characters`, async ({
```
Comment (sianford) [inline src/components/select/multi-select/multi-select.pw.tsx:72]: As above RE special characters and diacritics tests.

File: src/components/select/multi-select/multi-select.pw.tsx [hunk 11]
```diff
@@ -1,399 +1,29 @@
...
-      page,
-    }) => {
-      await mount(<MultiSelectComponent label={labelValue} />);
-
-      await expect(getDataElementByValue(page, "label")).toHaveText(labelValue);
-    });
-  });
-
-  testData.forEach((placeholderValue) => {
-    test(`should render placeholder using ${placeholderValue} special characters`, async ({
```
Comment (sianford) [inline src/components/select/multi-select/multi-select.pw.tsx:83]: As above RE special characters and diacritics tests.

File: src/components/select/multi-select/multi-select.pw.tsx [hunk 12]
```diff
@@ -1,399 +1,29 @@
...
-    await expect(commonDataElementInputPreview(page)).toBeFocused();
-  });
-
-  test("should check required prop", async ({ mount, page }) => {
-    await mount(<MultiSelectComponent required />);
-
-    await verifyRequiredAsteriskForLabel(page);
-  });
-
-  test("when maxWidth has no value it should render as 100%", async ({
```
Comment (sianford) [inline src/components/select/multi-select/multi-select.pw.tsx:151]: This test case is not tested in unit testing or in any test stories. The examples all contain a value for this prop. I think it should be added as an additional unit test.

File: src/components/select/multi-select/multi-select.pw.tsx [hunk 13]
```diff
@@ -1,399 +1,29 @@
...
-  }) => {
-    await mount(<MultiSelectComponent />);
-
-    await dropdownButton(page).click();
-    await expect(selectListWrapper(page)).toBeVisible();
-    await dropdownButton(page).click();
-    await expect(selectListWrapper(page)).toBeHidden();
-  });
-
-  test("should close the list with the Tab key", async ({ mount, page }) => {
```
Comment (sianford) [inline src/components/select/multi-select/multi-select.pw.tsx:210]: This keyboard functionality is not tested in unit testing. It should be added as a unit test if not in Playwright.

File: src/components/select/multi-select/multi-select.pw.tsx [hunk 14]
```diff
@@ -1,399 +1,29 @@
...
-    const selectListWrapperElement = selectListWrapper(page);
-    const selectInputElement = selectInput(page);
-    await dropdownButton(page).click();
-    await expect(selectListWrapperElement).toBeVisible();
-    await selectInputElement.press("Tab");
-    await expect(selectInputElement).toHaveAttribute("aria-expanded", "false");
-    await expect(selectListWrapperElement).toBeHidden();
-  });
-
-  test("should close the list with the Esc key", async ({ mount, page }) => {
```
Comment (sianford) [inline src/components/select/multi-select/multi-select.pw.tsx:222]: This keyboard functionality is not tested in unit testing. It should be added as a unit test if not in Playwright.

File: src/components/select/multi-select/multi-select.pw.tsx [hunk 15]
```diff
@@ -1,399 +1,29 @@
...
-      "title",
-      option1,
-    );
-    await expect(multiSelectPillByPosition(page, 1)).toHaveAttribute(
-      "title",
-      option2,
-    );
-  });
-
-  test("should check number of selected options are limited to 2", async ({
```
Comment (sianford) [inline src/components/select/multi-select/multi-select.pw.tsx:310]: This is not tested in unit testing or in any other automation. There's only an example in the Test stories. I think this should be left in.

File: src/components/select/multi-select/multi-select.pw.tsx [hunk 16]
```diff
@@ -1,399 +1,29 @@
...
-    }) => {
-      await mount(<MultiSelectComponent />);
-
-      await commonDataElementInputPreview(page).fill("foo");
-      await commonDataElementInputPreview(page).press(key);
-      await expect(page.getByText('No results for "foo"')).toBeVisible();
-    });
-  });
-
-  test("renders loader when loading prop is set to true", async ({
```
Comment (sianford) [inline src/components/select/multi-select/multi-select.pw.tsx:382]: This is not in unit testing and there is no Chromatic snapshot for this. I think it should be left in Playwright.

File: src/components/select/multi-select/multi-select.pw.tsx [hunk 17]
```diff
@@ -447,453 +77,6 @@ test.describe("MultiSelect component", () => {
...
-    await expect(secondPill).toHaveCSS("background-color", "rgba(0, 0, 0, 0)");
-  });
-
-  (
-    [
-      ["third", "Blue as the sky on a summer's day"],
-      ["fifth", "Green as the grass in a spring meadow"],
-    ] as const
-  ).forEach(([option, text]) => {
-    test(`should select ${option} list option and show pill with complete long text wrapped in the input`, async ({
```
Comment (sianford) [inline src/components/select/multi-select/multi-select.pw.tsx:685]: This isn't test in unit test and we don't have a Chromatic snapshot. I think it should be left in Playwright.

File: src/components/select/multi-select/multi-select.pw.tsx [hunk 18]
```diff
@@ -447,453 +77,6 @@ test.describe("MultiSelect component", () => {
...
-      />,
-    );
-
-    const pillElement = multiSelectPill(page);
-    await expect(pillElement).toHaveAttribute("title", "White");
-    await commonDataElementInputPreview(page).press("Backspace");
-    await expect(pillElement).toHaveCount(0);
-  });
-
-  test("should delete all selected pill options and leave list open", async ({
```
Comment (sianford) [inline src/components/select/multi-select/multi-select.pw.tsx:752]: This is not covered in unit testing. I think it should be left in Playwright.

File: src/components/select/multi-select/multi-select.pw.tsx [hunk 19]
```diff
@@ -447,453 +77,6 @@ test.describe("MultiSelect component", () => {
...
-    await mount(<MultiSelectComponent />);
-
-    await dropdownButton(page).click();
-    await expect(selectOptionByText(page, listOption)).toHaveAttribute(
-      "data-role",
-      "option1",
-    );
-  });
-
-  test("should render option data-element prop set to option1", async ({
```
Comment (sianford) [inline src/components/select/multi-select/multi-select.pw.tsx:821]: This prop is not checked in unit testing. A unit test should be added for this, like the `data-role` test.

File: src/components/select/simple-select/simple-select.pw.tsx [hunk 20]
```diff
@@ -44,178 +29,8 @@ import {
   checkAccessibility,
   positionOfElement,
 } from "../../../../playwright/support/helper";
-import { CHARACTERS } from "../../../../playwright/support/constants";
-
-const testData = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
```
Comment (sianford) [inline src/components/select/simple-select/simple-select.pw.tsx:49]: As with Filterable and Multi, these characters should still be tested as they aren't tested anywhere else.

File: src/components/select/simple-select/simple-select.pw.tsx [hunk 21]
```diff
@@ -44,178 +29,8 @@ import {
...
-const keyToTrigger = ["ArrowDown", "ArrowUp", "Space", "Home", "End"] as const;
 
 test.describe("SimpleSelect component", () => {
-  testData.forEach((labelValue) => {
-    test(`should render label using ${labelValue} special characters`, async ({
```
Comment (sianford) [inline src/components/select/simple-select/simple-select.pw.tsx:54]: As above RE special characters and diacritics.

File: src/components/select/simple-select/simple-select.pw.tsx [hunk 22]
```diff
@@ -44,178 +29,8 @@ import {
...
-      page,
-    }) => {
-      await mount(<SimpleSelectComponent label={labelValue} />);
-
-      await expect(getDataElementByValue(page, "label")).toHaveText(labelValue);
-    });
-  });
-
-  testData.forEach((placeholderValue) => {
-    test(`should render placeholder using ${placeholderValue} special characters`, async ({
```
Comment (sianford) [inline src/components/select/simple-select/simple-select.pw.tsx:65]: As above RE special characters and diacritics.

File: src/components/select/simple-select/simple-select.pw.tsx [hunk 23]
```diff
@@ -44,178 +29,8 @@ import {
...
-
-    const combobox = await page
-      .getByRole("combobox", { name: "simple select" })
-      .boundingBox();
-    const dropdownList = await page.getByRole("listbox").boundingBox();
-
-    expect(combobox?.y).toBeLessThan(dropdownList?.y ?? 0);
-  });
-
-  test("should close the list with the Tab key", async ({ mount, page }) => {
```
Comment (sianford) [inline src/components/select/simple-select/simple-select.pw.tsx:116]: There is not a unit test for this. An additional unit test should be added for this case.

File: src/components/select/simple-select/simple-select.pw.tsx [hunk 24]
```diff
@@ -44,178 +29,8 @@ import {
...
-    const selectListWrapperElement = selectListWrapper(page);
-    const selectInputElement = selectInput(page);
-    await selectText(page).click();
-    await expect(selectListWrapperElement).toBeVisible();
-    await selectInputElement.press("Tab");
-    await expect(selectInputElement).toHaveAttribute("aria-expanded", "false");
-    await expect(selectListWrapperElement).toBeHidden();
-  });
-
-  test("should close the list with the Esc key", async ({ mount, page }) => {
```
Comment (sianford) [inline src/components/select/simple-select/simple-select.pw.tsx:128]: There is not a unit test for this. An additional unit test should be added for this case.

File: src/components/select/simple-select/simple-select.pw.tsx [hunk 25]
```diff
@@ -44,178 +29,8 @@ import {
...
-
-      await selectText(page).click();
-      await selectOptionByText(page, option).click();
-      await expect(getDataElementByValue(page, "input")).toHaveValue(option);
-      await expect(selectInput(page)).toHaveAttribute("aria-expanded", "false");
-      await expect(selectListWrapper(page)).toBeHidden();
-    });
-  });
-
-  test("should render an option that wraps onto more than one line correctly", async ({
```
Comment (sianford) [inline src/components/select/simple-select/simple-select.pw.tsx:181]: This isn't being tested anywhere else. I think it should be left in.

File: src/components/select/simple-select/simple-select.pw.tsx [hunk 26]
```diff
@@ -44,178 +29,8 @@ import {
...
-    await selectOptionByText(page, optionValue11).scrollIntoViewIfNeeded();
-    await expect(selectOptionByText(page, optionValue8)).toBeInViewport();
-    await expect(
-      selectOptionByText(page, optionValue9).nth(1),
-    ).toBeInViewport();
-    await expect(selectOptionByText(page, optionValue10)).toBeInViewport();
-    await expect(selectOptionByText(page, optionValue11)).toBeInViewport();
-  });
-
-  test("renders loader when isLoading prop is set to true", async ({
```
Comment (sianford) [inline src/components/select/simple-select/simple-select.pw.tsx:204]: This isn't being tested anywhere else. I think it should be left in. There is no unit test for the isLoading prop either.

File: src/components/select/simple-select/simple-select.pw.tsx [hunk 27]
```diff
@@ -268,55 +83,6 @@ test.describe("SimpleSelect component", () => {
     expect(newScrollPosition).toBeCloseTo(scrollPosition, 1);
   });
 
-  test("keyboard navigation should work correctly in multicolumn mode and ensure the selected option is visible", async ({
```
Comment (sianford) [inline src/components/select/simple-select/simple-select.pw.tsx:271]: This isn't being tested anywhere else. I think it should be left in. Unit tests only cover opening and closing of the list with arrow keys.

File: src/components/select/simple-select/simple-select.pw.tsx [hunk 28]
```diff
@@ -376,471 +142,6 @@ test.describe("SimpleSelect component", () => {
...
-
-    await selectText(page).click();
-    await expect(selectListWrapper(page)).toBeVisible();
-    await expect(selectListCustomChild(page, 1).locator("..")).toHaveAttribute(
-      "data-component",
-      "option",
-    );
-  });
-
-  test("should render custom option data-role prop set to option", async ({
```
Comment (sianford) [inline src/components/select/simple-select/simple-select.pw.tsx:581]: This prop isn't being tested anywhere. I think it should have a unit test.

File: src/components/select/simple-select/simple-select.pw.tsx [hunk 29]
```diff
@@ -376,471 +142,6 @@ test.describe("SimpleSelect component", () => {
...
-
-    await selectText(page).click();
-    await expect(selectListWrapper(page)).toBeVisible();
-    await expect(selectListCustomChild(page, 1).locator("..")).toHaveAttribute(
-      "data-role",
-      "option",
-    );
-  });
-
-  test("should render custom option data-element prop set to option", async ({
```
Comment (sianford) [inline src/components/select/simple-select/simple-select.pw.tsx:595]: This prop isn't being tested anywhere. I think it should have a unit test.

File: src/components/select/simple-select/simple-select.pw.tsx [hunk 30]
```diff
@@ -376,471 +142,6 @@ test.describe("SimpleSelect component", () => {
...
-    await expect(wrapperElement).toBeVisible();
-  });
-
-  (
-    [
-      ["top", "300px", "0px", "200px", "20px"],
-      ["bottom", "0px", "0px", "0px", "20px"],
-    ] as const
-  ).forEach(([position, top, bottom, left, right]) => {
-    test(`should render list in ${position} position when margins are top ${top}, bottom ${bottom}, left ${left} and right ${right}`, async ({
```
Comment (sianford) [inline src/components/select/simple-select/simple-select.pw.tsx:628]: All of these margin props are not tested anywhere else. I think this should be left in.

File: src/components/select/simple-select/simple-select.pw.tsx [hunk 31]
```diff
@@ -376,471 +142,6 @@ test.describe("SimpleSelect component", () => {
...
-    });
-  });
-
-  (
-    [
-      ["top", "0px", "0px", "0px", "20px"],
-      ["bottom", "600px", "0px", "0px", "20px"],
-    ] as [SimpleSelectProps["listPlacement"], string, string, string, string][]
-  ).forEach(([position, top, bottom, left, right]) => {
-    test(`should flip list to opposite position when there is not enough space to render in ${position} position`, async ({
```
Comment (sianford) [inline src/components/select/simple-select/simple-select.pw.tsx:658]: Not all positions are tested anywhere else. I think this is a good Chromatic candidate, as in have an example of each in the Chromatic story.

File: src/components/select/simple-select/simple-select.pw.tsx [hunk 32]
```diff
@@ -376,471 +142,6 @@ test.describe("SimpleSelect component", () => {
...
-        position,
-      );
-      await expect(listElement).toBeVisible();
-    });
-  });
-
-  (
-    ["top", "top-start", "top-end"] as SimpleSelectProps["listPlacement"][]
-  ).forEach((position) => {
-    test(`should render list with expected box-shadow when listPosition is ${position}`, async ({
```
Comment (sianford) [inline src/components/select/simple-select/simple-select.pw.tsx:742]: This is not tested anywhere else. If the above placement examples are added into the Chromatic story then this would be covered there.

File: src/components/select/filterable-select/filterable-select.pw.tsx (hunk 1 above)
Comment (paulrobinson88) [reply src/components/select/filterable-select/filterable-select.pw.tsx:46]: the thinking was to have the special characters covered in the simple-select but it does make sense to keep in all 3 select versions in case of specific changes around this functionality in just one, added back in.

File: src/components/select/filterable-select/filterable-select.pw.tsx (hunk 6 above)
Comment (paulrobinson88) [reply src/components/select/filterable-select/filterable-select.pw.tsx:177]: feel like the below may be sufficient but will check. https://github.com/Sage/carbon/blob/fee280c4a3c8f109b418b545e4e71fd15773e660/src/components/select/filterable-select/filterable-select.test.tsx#L1968-L1979

File: src/components/select/filterable-select/filterable-select.stories.tsx (hunk 8 above)
Comment (paulrobinson88) [reply src/components/select/filterable-select/filterable-select.stories.tsx:30]: yep this was intended since the dropdown isn't open I don't think they add much value and anything we do need is covered in the interactions tests. The 'No results' snapshot seems ok to me in the library, will check when the build runs on this PR though. 
Screenshot 2026-07-01 at 14 17 45

File: src/components/select/multi-select/multi-select.pw.tsx (hunk 12 above)
Comment (paulrobinson88) [reply src/components/select/multi-select/multi-select.pw.tsx:151]: Like the one above, i'll check on this as I believe the default would be applied when no value is passed (treat as falsy), which would be to apply maxWidth 100% which we have covered so it may be redundant to check for this also.

File: src/components/select/multi-select/multi-select.pw.tsx (hunk 17 above)
Comment (paulrobinson88) [reply src/components/select/multi-select/multi-select.pw.tsx:685]: good spot, i've added this to the chromatic snapshot story so it's captured.

File: src/components/select/simple-select/simple-select.pw.tsx (hunk 29 above)
Comment (paulrobinson88) [reply src/components/select/simple-select/simple-select.pw.tsx:595]: thought this was being tested in Jest but appears to just be part of styledsystem, added in

File: src/components/select/simple-select/simple-select.pw.tsx (hunk 31 above)
Comment (paulrobinson88) [reply src/components/select/simple-select/simple-select.pw.tsx:658]: looks like this is covered in `filterable-select` so not sure if it's needed in `simple-select` also. Will check if we would need it in here also. https://github.com/Sage/carbon/blob/7fa91f14933099b8d9114ddd7b59ae905d901bc1/src/components/select/filterable-select/filterable-select.pw.tsx#L236-L237

File: src/components/select/simple-select/simple-select.pw.tsx (hunk 32 above)
Comment (paulrobinson88) [reply src/components/select/simple-select/simple-select.pw.tsx:742]: added this back in as I think it's likely more straightforward than adding an interaction and checking against each box-shadow.

File: src/components/select/filterable-select/filterable-select.pw.tsx (hunk 6 above)
Comment (paulrobinson88) [reply src/components/select/filterable-select/filterable-select.pw.tsx:177]: checked with the team, as it's effectively falsy this is covered when the max-width isn't set.

File: src/components/select/multi-select/multi-select.pw.tsx (hunk 12 above)
Comment (paulrobinson88) [reply src/components/select/multi-select/multi-select.pw.tsx:151]: checked with the team, as it's effectively falsy this is covered when the max-width isn't set.

File: src/components/select/filterable-select/filterable-select.pw.tsx (hunk 6 above)
Comment (sianford) [reply src/components/select/filterable-select/filterable-select.pw.tsx:177]: They're different test cases though. Passing a null value for an existing prop is not the same as not passing the prop at all. The assertions are on different props. In the PW test, when `maxWidth=""` it asserts that `max-width=100%`. However, in the unit test, when `maxWidth` prop is not passed, it asserts that `width=100%` and specifically checks `max-width=!100%`.

File: src/components/select/filterable-select/filterable-select.stories.tsx (hunk 8 above)
Comment (sianford) [reply src/components/select/filterable-select/filterable-select.stories.tsx:30]: I have since checked this, and it is correct so you can ignore this one.

File: src/components/select/multi-select/multi-select.pw.tsx (hunk 12 above)
Comment (sianford) [reply src/components/select/multi-select/multi-select.pw.tsx:151]: As above, testing for a null value is not the same as testing for when a prop is not passed. `prop={undefined}` (or not passing the prop) means the prop is undefined. `prop=""` means the prop is an empty string. They're asserting opposite things on the `max-width` style.

File: src/components/select/filterable-select/filterable-select.pw.tsx (hunk 6 above)
Comment (paulrobinson88) [reply src/components/select/filterable-select/filterable-select.pw.tsx:177]: added this back in

File: src/components/select/multi-select/multi-select.pw.tsx (hunk 12 above)
Comment (paulrobinson88) [reply src/components/select/multi-select/multi-select.pw.tsx:151]: added back in

File: src/components/select/filterable-select/filterable-select.pw.tsx (hunk 1 above)
Comment (edleeks87) [reply src/components/select/filterable-select/filterable-select.pw.tsx:46]: these don't really need to be pw tests, unit testing will cover this or an interactive story, it's non-blocking for me though as it's not a huge deal as long as it's tested

File: src/components/select/filterable-select/filterable-select.pw.tsx (hunk 6 above)
Comment (edleeks87) [reply src/components/select/filterable-select/filterable-select.pw.tsx:177]: This is a pointless test, no one will ever pass an empty string here, we're covering an edge case that would be a bug in the implementation code not the component's

File: src/components/select/multi-select/multi-select.pw.tsx (hunk 12 above)
Comment (edleeks87) [reply src/components/select/multi-select/multi-select.pw.tsx:151]: same comment as above no one is going to pass `''` here and it not be a bug in their code

File: src/components/select/multi-select/multi-select.pw.tsx [hunk 33]
```diff
@@ -159,241 +73,6 @@ test.describe("MultiSelect component", () => {
...
-      "title",
-      option1,
-    );
-    await expect(multiSelectPillByPosition(page, 1)).toHaveAttribute(
-      "title",
-      option2,
-    );
-  });
-
-  test("should check number of selected options are limited to 2", async ({
```
Comment (edleeks87) [inline src/components/select/multi-select/multi-select.pw.tsx:310]: this is a test of an implementation rather than specific component behaviour, unless it was added to cover a regression for a previously raised issue it's not what we should be testing in carbon imo

File: src/components/select/filterable-select/filterable-select.pw.tsx (hunk 1 above)
Comment (sianford) [reply src/components/select/filterable-select/filterable-select.pw.tsx:46]: Unit testing only goes as far as alphanumeric. The reason we had these in PW at all was because of this.

File: src/components/select/multi-select/multi-select.pw.tsx (hunk 33 above)
Comment (sianford) [reply src/components/select/multi-select/multi-select.pw.tsx:310]: That's not what this test is. It's testing that when two particular options were selected, two pills for those options are shown in the Select field. I think what happened at some point before is there was a bug whereby options in the list between the ones selected were also added, so this checks that only options 1 and 3 are added, not 2 as well.

Comment (edleeks87) [reply src/components/select/multi-select/multi-select.pw.tsx:310]: It seems pretty specific when we likely have other tests that satisfy this enough

File: src/components/select/filterable-select/filterable-select.pw.tsx (hunk 1 above)
Comment (edleeks87) [reply src/components/select/filterable-select/filterable-select.pw.tsx:46]: add them to the unit tests?

Comment (sianford) [reply src/components/select/filterable-select/filterable-select.pw.tsx:46]: Fine with that.

File: src/components/select/multi-select/multi-select.pw.tsx (hunk 9 above)
Comment (sianford) [reply src/components/select/multi-select/multi-select.pw.tsx:52]: Or prob better placed in unit testing, @paulrobinson88, as Ed suggested for Filterable above.

File: src/components/select/multi-select/multi-select.pw.tsx (hunk 15 above)
Comment (sianford) [reply src/components/select/multi-select/multi-select.pw.tsx:310]: @paulrobinson88 you can take this out again if you like based on Ed's comments around it testing an implementation.

**Review by sianford — DISMISSED** [review]
Nice work, @paulrobinson88. It was a massive component to take on. 3 in 1, really.

**Review by edleeks87 — DISMISSED** [review]

**Review by mihai-albu-sage — DISMISSED** [review]

**Review by sianford — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]
